Automattic/wp-calypso
# SetAsPrimary: fix error notice and rewrite Redux usage to hooks

Fixes a bug in `SetPrimary` where a success notice would be shown even on error, replacing the `showUpdatePrimaryDomainSuccessNotice` action with the correct `showUpdatePrimaryDomainErrorNotice`.

Then the PR removes the `connect` calls and moves the code inside the component into `useSelector` calls. That also greatly simplifies the types because we don't need to declare types for connected props any more.

Part of #64069.

**How to test:**
Go to settings for a domain that's not a primary one and test the "Set as primary" accordion. Verify that you can set the domain as primary from there.
